File: loop_and_tail/src/fib_tail.py
import sys


def fib(n):
    def fibtail(n, second, first):
        if n < 3:
            return first + second
        return fibtail(n - 1, first, first + second)
    return fibtail(n, 0, 1)

if __name__ == '__main__':
    limit = 1000000000
    sys.setrecursionlimit(limit)

    n = int(sys.argv[1])
    print("res = {}".format(fib(n)))

File: loop_and_tail/src/fib_tail.erl
-module(fib_tail).
-export([main/0]).

fib_tail(N, Second, First) when N < 3 -> Second + First;
fib_tail(N, Second, First) -> fib_tail(N - 1, First, Second + First).

fib(N) -> fib_tail(N, 0, 1).

main() ->
    {ok, Arg} = io:read(""),
    io:format("res: ~p\n", [fib(Arg)]).
